build-packages/webpack-i18n-runtime/src/component/I18n/I18n.component.js

```diff
@@ -0,0 +1,41 @@
+import PropTypes from 'prop-types';
+import { PureComponent } from 'react';
+
+import i18n from '../../util/i18n';
+
+/** @namespace RuntimeI18n/Component/I18n/Component/I18nComponent */
+export class I18nComponent extends PureComponent {
+    static propTypes = {
+        children: PropTypes.node.isRequired
+    };
+
+    // eslint-disable-next-line @scandipwa/scandipwa-guidelines/only-render-in-component
+    constructor(props) {
+        super(props);
+
+        i18n.init(this.forceUpdate.bind(this));
+    }
+
+    render() {
+        const { children } = this.props;
+        const currentLocale = i18n.getCurrentLocale();
+
+        // Do not render anything while the translation is loading
+        if (i18n.isLoading) {
+            return null;
+        }
```

Comment:
Possibly, we can still start rendering? Maybe a plugin to Text placeholder might help, where while loading, you would keep placeholders showing?

Reply:
During the initial page load, the default locale chunk will be preloaded a long time before this is executed, hence this will not impact the performance.

On locale change, the intended behavior is:
1. Load the chunk, wait until loaded
2. Trigger `forceUpdate` of this component, `i18n.isLoading === false` at this moment
3. The application gets rerendered

This check is a measure to prevent the English locale from blinking during the initial page load in case the default locale chunk is not preloaded.

Having text placeholders everywhere would be nice, but the current implementation of the `__(...)` function returns text, not JSX, hence - it cannot show text placeholder. We would need to refactor the application and insert text placeholders manually everywhere, which would add an additional level of API to the package and if we decide to do so, I vote for reconsidering the HOC option alongside this change.